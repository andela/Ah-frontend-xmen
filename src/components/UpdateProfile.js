import React, { Component } from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class UpdateProfile extends Component {
  render() {
    const { onSubmit, onChange, profile } = this.props;
    return (
      <div>
        <div className="container edit-container">
          <h5 className="mt-5">Edit Profile</h5>
          <form onSubmit={onSubmit} className="mt-2 mb-5">
            <div className="form-group">
              <label htmlFor="fname">First Name</label>
              <input type="text" id="fname" name="first_name" placeholder="Your name.." value={profile.first_name} onChange={onChange} className="form-control" />
            </div>
            <div className="form-group">
              <label htmlFor="lname" id="lname">Last Name</label>
              <input type="text" id="lname" name="last_name" placeholder="Your last name.." value={profile.last_name} onChange={onChange} className="form-control" />
            </div>
            <div>
              <label htmlFor="bio">Bio</label>
              <textarea name="bio" className="form-control" value={profile.bio} onChange={onChange} rows="5" maxLength="360">Your Bio</textarea>
            </div>
            <button type="submit" className="btn button-primary mt-3" id="formsubmit">Submit</button>
          </form>
        </div>
      </div>
    );
  }
}

UpdateProfile.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};
export default UpdateProfile;
