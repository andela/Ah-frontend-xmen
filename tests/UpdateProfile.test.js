import React from 'react';
import { shallow } from 'enzyme';
import UpdateProfile from '../src/components/UpdateProfile';

describe('update profile view', () => {
  it('should render without crashing', () => {
    const props = {
      onSubmit: jest.fn(),
      onChange: jest.fn(),
      profile: {
        first_name: 'firstname',
        last_name: 'lastname',
        bio: 'today',
      },
    };
    const wrapper = shallow(<UpdateProfile {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
