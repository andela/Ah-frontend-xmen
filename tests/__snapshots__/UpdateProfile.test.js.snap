// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update profile view should render without crashing 1`] = `ShallowWrapper {}`;
